app/services/irs_attempts_api/tracker.rb

```diff
@@ -1,26 +1,27 @@
 module IrsAttemptsApi
   class Tracker
     attr_reader :session_id, :enabled_for_session, :request, :user, :sp, :cookie_device_uuid,
-                :sp_request_uri
+                :sp_request_uri, :analytics
 
     def initialize(session_id:, request:, user:, sp:, cookie_device_uuid:,
-                   sp_request_uri:, enabled_for_session:)
+                   sp_request_uri:, enabled_for_session:, analytics:)
       @session_id = session_id # IRS session ID
       @request = request
       @user = user
       @sp = sp
       @cookie_device_uuid = cookie_device_uuid
       @sp_request_uri = sp_request_uri
       @enabled_for_session = enabled_for_session
+      @analytics = analytics
     end
 
     def track_event(event_type, metadata = {})
-      return unless enabled?
+      return if !enabled? && !IdentityConfig.store.irs_attempt_api_payload_size_logging_enabled
```

Comment:
This is perhaps the only case where `unless ... || ...` would read cleanly to me since we're negating both.

That's generally seen as a cardinal sin, though, so I'm fine with this as-is.

Reply:
It's my personal vendetta against `unless`. Another idea was an explicit negation/or. What do you think of this?

```suggestion
      return if !(enabled? || IdentityConfig.store.irs_attempt_api_payload_size_logging_enabled)
```

Reply:
going to leave it as is for the time being but open to changing later if this confuses folks!

Reply:
My brain has a hard time with `!( ... || ... )` / `!( ... && ... )` doing mental De Morgan's to decipher the logic, so I'm happier with it as is, and would have also been fine with `unless` (I don't share the same vendetta 😄 ).